### PR TITLE
allow global configuration of default storage classes

### LINKED ISSUE #905: Enhancement: Add build and cache storage class flags to controller
Currently, build and cache storage classes cannot be configured globally.

This has the unfortunate side-effect that if the default storage class on your cluster isn't adequate (e.g. doesn't support `AccessMode: ReadWriteMany`), the users of `brig project create` are forced down an advanced configuration path that they otherwise may not want to mess with.

By exposing these flags, we can, in turn, expose these options through Brigade's Helm chart, thus putting default build and cache storage class selection in the hands of whoever administers the Brigade installation instead of individual project owners.

### PR DESCRIPTION
**Pull request by krancour**
Fixes #905

### CODE REVIEW

**Comment by krancour** [comment]
@vdice I added the unit tests you suggested. JS/TS is not my strong suit. lmk if these look ok.

File: brigade-worker/test/k8s.ts [hunk 1]
```diff
@@ -556,6 +556,119 @@ describe("k8s", function() {
...
+            assert.equal(pvc.spec.storageClassName, jr.project.kubernetes.cacheStorageClass)
+          });
+        });
+        context("when the cache storage class is not overridden at the project level", () => {
+          beforeEach(function() {
+            jr.project.kubernetes.cacheStorageClass = "";
+          });
+          it("it falls back on the cluster default", () => {
+            let pvc = jr['cachePVC']()
+            // Undefined means k8s will use the cluser's default storage class
```
Comment (vdice) [inline brigade-worker/test/k8s.ts]: typo: `cluser's` (same on line... 666 😅 )

Comment (krancour) [reply brigade-worker/test/k8s.ts]: Good eye. Fixing.

Comment (krancour) [reply brigade-worker/test/k8s.ts]: Fixed

**Pull request merged**